tasks/Functions-client/deployAutoClient.js

```diff
@@ -36,42 +37,48 @@ task("functions-deploy-auto-client", "Deploys the AutomatedFunctionsConsumer con
       throw Error("Gas limit must be less than or equal to 300,000")
     }
 
-    console.log(`Deploying AutomatedFunctionsConsumer contract to ${network.name}`)
-
     console.log("\n__Compiling Contracts__")
     await run("compile")
 
+    console.log(`Deploying AutomatedFunctionsConsumer contract to ${network.name}`)
+
+    const functionsRouterAddress = networks[network.name]["functionsRouter"]
+    const donIdBytes32 = hre.ethers.utils.formatBytes32String(networks[network.name]["donId"])
+    const signer = await ethers.getSigner()
+    const linkTokenAddress = networks[network.name]["linkToken"]
+    const txOptions = { confirmations: networks[network.name].confirmations }
+
+    const subId = taskArgs.subid
+
     const autoClientContractFactory = await ethers.getContractFactory("AutomatedFunctionsConsumer")
     const autoClientContract = await autoClientContractFactory.deploy(
-      networks[network.name]["functionsOracleProxy"],
-      taskArgs.subid,
+      functionsRouterAddress,
+      donIdBytes32,
+      subId,
       taskArgs.gaslimit,
       taskArgs.interval
     )
 
     console.log(`\nWaiting 1 block for transaction ${autoClientContract.deployTransaction.hash} to be confirmed...`)
     await autoClientContract.deployTransaction.wait(1)
 
-    await addClientConsumerToSubscription(taskArgs.subid, autoClientContract.address)
+    const consumerAddress = autoClientContract.address
 
-    taskArgs.contract = autoClientContract.address
+    const sm = new SubscriptionManager({ signer, linkTokenAddress, functionsRouterAddress })
+    await sm.initialize()
 
-    await setAutoRequest(autoClientContract.address, taskArgs)
+    console.log(`\nAdding ${consumerAddress} to subscription ${subId}...`)
+    const addConsumerTx = await sm.addConsumer({ subId, consumerAddress, txOptions })
+    console.log(`\nAdded consumer contract ${consumerAddress} in Tx: ${addConsumerTx.transactionHash}`)
 
```

Comment:
I am wondering if it makes sense to set the request here too upon deployment.  Basically, instead of users needing to deploy and then run set-auto-request, they could just deploy and the request would be set automatically.  This could be accomplished by building the CBOR bytes in this file and adding the requestCBOR as a constructor arg.

Reply:
Else... Maybe we do require users to run `set-auto-request`, but then we don't need to pass the subId, gasLimit or interval as constructor args.

Reply:
Have kept this separate after a bit of deliberation as i feel that combining the two steps into the deploy step will "hide" and important step needed for the dev to understand the lifecylcle/flow.